src/node/internal/internal_fs_sync.ts

```diff
@@ -0,0 +1,1066 @@
+    } = offsetOrOptions;
+    actualOffset = offset;
+    actualLength = length;
+    actualPosition = position;
+  }
+  // Handle the case where the third argument is a number (offset)
+  else if (typeof offsetOrOptions === 'number') {
+    actualOffset = offsetOrOptions;
+    actualLength = length ?? buffer.byteLength - actualOffset;
+    actualPosition = position;
+  } else {
+    throw new ERR_INVALID_ARG_TYPE(
+      'offset',
+      ['number', 'object'],
+      offsetOrOptions
+    );
+  }
+
+  validateUint32(actualOffset, 'offset');
+  validateUint32(actualLength, 'length');
+  validatePosition(actualPosition, 'position');
+
+  // The actualOffset plus actualLength must not exceed the buffer's byte length.
+  if (actualOffset + actualLength > buffer.byteLength) {
+    throw new ERR_INVALID_ARG_VALUE('offset', actualOffset, 'out of bounds');
+  }
+
+  if (actualLength === 0 || buffer.byteLength === 0) {
+    return 0;
+  }
+
+  return readvSync(
+    fd,
+    [Buffer.from(buffer.buffer, actualOffset, actualLength)],
+    actualPosition
+  );
+}
+
+export function readvSync(
+  fd: number,
+  buffers: ArrayBufferView[],
+  position: number | bigint | null = null
+): number {
+  fd = getValidatedFd(fd);
+  validateBufferArray(buffers);
+
+  if (
+    position != null &&
+    typeof position !== 'number' &&
+    typeof position !== 'bigint'
+  ) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'position',
+      ['null', 'number', 'bigint'],
+      position
+    );
+  }
+
+  if (buffers.length === 0) {
+    return 0;
+  }
+  return cffs.read(fd, buffers, { position });
+}
+
+// TODO: Implement fs.realpathSync.native
+export function realpathSync(
+  p: FilePath,
+  options: ReadLinkSyncOptions = {}
+): string | Buffer {
+  validateObject(options, 'options');
+  const { encoding = 'utf8' } = options;
+  validateEncoding(encoding, 'options.encoding');
+  const dest = Buffer.from(
+    cffs.readLink(normalizePath(p), { failIfNotSymlink: false })
+  );
+  if (typeof encoding === 'string') {
+    return dest.toString(encoding);
+  }
+  return dest;
+}
+
+realpathSync.native = realpathSync;
+
+export function renameSync(src: FilePath, dest: FilePath): void {
+  cffs.renameOrCopy(normalizePath(src), normalizePath(dest), { copy: false });
+}
+
+export type RmdirSyncOptions = {
+  maxRetries?: number;
+  recursive?: boolean;
+  retryDelay?: number;
+};
+
+export function rmdirSync(
+  path: FilePath,
+  options: RmdirSyncOptions = {}
+): void {
+  validateObject(options, 'options');
+  const { maxRetries = 0, recursive = false, retryDelay = 0 } = options;
+  // We do not implement the maxRetries or retryDelay options in any meaningful
+  // way. We just validate them.
+  validateUint32(maxRetries, 'options.maxRetries');
+  validateBoolean(recursive, 'options.recursive');
+  validateUint32(retryDelay, 'options.retryDelay');
+
+  cffs.rm(normalizePath(path), { recursive, force: false, dironly: true });
+}
+
+export type RmSyncOptions = {
```

Comment:
This can just be `RmOptions`, or it can be `import type { RmOptions } from 'node:fs'`.

Again, partial support would be good (`force?: boolean | undefined` etc).